storage/client_event_test.go

```diff
@@ -0,0 +1,245 @@
+		fmt.Printf("proto.StoreID(%d): []string{\n", int64(id))
+		for _, evt := range feed {
+			fmt.Printf("\t\t\"%s\",\n", evt)
+		}
+		fmt.Printf("},\n")
+	}
+}
+
+// TestMultiStoreEventFeed verifies that events on multiple stores are properly
+// recieved by a single event reader.
+func TestMultiStoreEventFeed(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	// Create a multiTestContext which publishes all store events to the given
+	// feed.
+	ser := &storeEventReader{}
+	feed := &util.Feed{}
+	mtc := &multiTestContext{
+		feed: feed,
+	}
+
+	// Start reading events from the feed before starting the stores.
+	readStopper := util.NewStopper()
+	sub := feed.Subscribe()
+	readStopper.RunWorker(func() {
+		ser.readEvents(sub)
+	})
+
+	mtc.Start(t, 3)
+	defer mtc.Stop()
+
+	// Replicate the default range.
+	raftID := int64(1)
+	mtc.replicateRange(raftID, 0, 1, 2)
+
+	// Add some data in a transaction
+	err := mtc.db.RunTransaction(nil, func(txn *client.Txn) error {
+		return txn.Run(
+			client.PutCall(proto.Key("a"), []byte("asdf")),
+			client.PutCall(proto.Key("c"), []byte("jkl;")),
+		)
+	})
+	if err != nil {
+		t.Fatalf("error putting data to db: %s", err.Error())
+	}
+
+	// AdminSplit in between the two ranges.
+	err = mtc.db.Run(
+		client.Call{
+			Args: &proto.AdminSplitRequest{
+				RequestHeader: proto.RequestHeader{
+					Key: proto.Key("b"),
+				},
+				SplitKey: proto.Key("b"),
+			},
+			Reply: &proto.AdminSplitResponse{},
+		},
+	)
+	if err != nil {
+		t.Fatalf("error splitting initial: %s", err.Error())
+	}
+
+	// AdminSplit an empty range at the end of the second range.
+	err = mtc.db.Run(
+		client.Call{
+			Args: &proto.AdminSplitRequest{
+				RequestHeader: proto.RequestHeader{
+					Key: proto.Key("z"),
+				},
+				SplitKey: proto.Key("z"),
+			},
+			Reply: &proto.AdminSplitResponse{},
+		},
+	)
+	if err != nil {
+		t.Fatalf("error splitting second range: %s", err.Error())
+	}
+
+	// AdminMerge the empty range back into the second range.
+	err = mtc.db.Run(
+		client.Call{
+			Args: &proto.AdminMergeRequest{
+				RequestHeader: proto.RequestHeader{
+					Key: proto.Key("c"),
+				},
+			},
+			Reply: &proto.AdminMergeResponse{},
+		},
+	)
+	if err != nil {
+		t.Fatalf("error merging final range: %s", err.Error())
+	}
+
+	// Add an additional put through the system and wait for all
+	// replicas to receive it.
+	err = mtc.db.Run(
+		client.IncrementCall(proto.Key("aa"), 5),
+	)
+	if err != nil {
+		t.Fatalf("error putting data to db: %s", err.Error())
+	}
+	util.SucceedsWithin(t, time.Second, func() error {
+		for _, eng := range mtc.engines {
+			val, err := engine.MVCCGet(eng, proto.Key("aa"), mtc.clock.Now(), true, nil)
+			if err != nil {
+				return err
+			}
+			if a, e := val.GetInteger(), int64(5); a != e {
+				return util.Errorf("expected aa = %d, got %d", e, a)
+			}
+		}
+		return nil
+	})
+
+	// Close feed and wait for reader to receive all events.
+	feed.Close()
+	readStopper.Stop()
+	<-readStopper.IsStopped()
```

Comment:
This is redundant; `Stop` does not return until everything has stopped. `IsStopped` is only needed if you need to wait for this event from a different goroutine from the one that calls `Stop` (which is rare but we do occasionally).